aspnetcore/fundamentals/file-providers.md

```diff
@@ -1,148 +1,261 @@
+
+[!code-csharp[](file-providers/samples/2.x/FileProviderSample/Startup.cs?name=snippet1)]
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.0"
+
+In the sample app, a `PhysicalFileProvider` and an `EmbeddedFileProvider` provide files to a `CompositeFileProvider` registered in the app's service container:
+
+[!code-csharp[](file-providers/samples/1.x/FileProviderSample/Startup.cs?name=snippet1)]
+
+::: moniker-end
+
+## Watch for changes
 
-## Globbing patterns
+The [IFileProvider.Watch](/dotnet/api/microsoft.extensions.fileproviders.ifileprovider.watch) method provides a scenario to watch one or more files or directories for changes. `Watch` accepts a path string, which can use [glob patterns](#glob-patterns) to specify multiple files. `Watch` returns an [IChangeToken](/dotnet/api/microsoft.extensions.primitives.ichangetoken). The change token exposes:
 
-File system paths use wildcard patterns called *globbing patterns*. These simple patterns can be used to specify groups of files. The two wildcard characters are `*` and `**`.
+* [HasChanged](/dotnet/api/microsoft.extensions.primitives.ichangetoken.haschanged): A property that can be inspected to determine if a change has occurred.
+* [RegisterChangeCallback](/dotnet/api/microsoft.extensions.primitives.ichangetoken.registerchangecallback): Called when changes are detected to the specified path string. Each change token only calls its associated callback in response to a single change. To enable constant monitoring, use a [TaskCompletionSource](/dotnet/api/system.threading.tasks.taskcompletionsource-1) (shown below) or recreate `IChangeToken` instances in response to changes.
 
-**`*`**
+In the sample app, the *WatchConsole* console app is configured to display a message whenever a text file is modified:
 
-   Matches anything at the current folder level, or any filename, or any file extension. Matches are terminated by `/` and `.` characters in the file path.
+::: moniker range=">= aspnetcore-2.0"
 
-<strong><code>**</code></strong>
+[!code-csharp[](file-providers/samples/2.x/WatchConsole/Program.cs?name=snippet1&highlight=1-2,16,19-20)]
 
-   Matches anything across multiple directory levels. Can be used to recursively match many files within a directory hierarchy.
+::: moniker-end
 
-### Globbing pattern examples
+::: moniker range="< aspnetcore-2.0"
 
-**`directory/file.txt`**
+[!code-csharp[](file-providers/samples/1.x/WatchConsole/Program.cs?name=snippet1&highlight=1-2,16,19-20)]
 
-   Matches a specific file in a specific directory.
+::: moniker-end
 
-**<code>directory/*.txt</code>**
+Some file systems, such as Docker containers and network shares, may not reliably send change notifications. Set the `DOTNET_USE_POLLINGFILEWATCHER` environment variable to `1` or `true` to poll the file system for changes every four seconds.
```

Comment:
Is this 4-second interval configurable?

Reply:
@danroth27 I can't find an online reference for your remark [here](https://github.com/aspnet/Docs/issues/206#issue-73200945) on the four second polling value.